tjreports/site/models/reports.php

```diff
@@ -559,13 +559,17 @@ protected function populateState($ordering = '', $direction = 'ASC')
 		if (empty($colToshow))
 		{
 			$reportParams = $this->getReportParams($reportId);
-			$colToshow    = (array) $reportParams->get("colToshow");
-			$piiColumns   = (array) $reportParams->get("piiColumns");
-			$piiColumns   = array_flip($piiColumns);
 
-			if (!empty($piiColumns))
+			if(!empty($reportParams['data']))
 			{
-				$colToshow = (object) array_diff_key($colToshow, $piiColumns);
+				$colToshow    = (array) $reportParams->get("colToshow");
```

Comment:
Better to declare these variables as empty arrays at the beginning of the function right ?